operator/src/main/java/oracle/kubernetes/operator/JobWatcher.java

```diff
@@ -283,4 +302,39 @@ public NextAction onSuccess(
           });
     }
   }
+
+  static class DeadlineExceededException extends Exception {
+    final V1Job job;
+
+    public DeadlineExceededException(V1Job job) {
+      super();
+      this.job = job;
+    }
+
+    public String toString() {
+      StringBuilder sb = new StringBuilder("Job ")
+          .append(job.getMetadata().getName()).append(" failed due to reason: DeadlineExceeded.")
```

Comment:
Sigh... Now we need a change as this needs to be internationalized.

Reply:
You can use code like here to format using the a message in our resource bundle: https://docs.oracle.com/javase/tutorial/i18n/format/messageFormat.html.  Our bundle is Operator.properties and the keys are in MessageKeys.java.